components/org.wso2.carbon.identity.api.server.email.template/org.wso2.carbon.identity.rest.api.server.email.template.v1/src/main/java/org/wso2/carbon/identity/rest/api/server/email/template/v1/core/ServerEmailTemplatesService.java

```diff
@@ -392,26 +414,66 @@ private void addEmailTemplateToTheSystem(String templateTypeDisplayName, EmailTe
-                emailTemplateType.setId(templateTypeId);
-                // Set location.
-                emailTemplateType.setSelf(getTemplateTypeLocation(templateTypeId));
+        for (String templateType : emailTemplateTypes) {
+
+            EmailTemplateTypeWithoutTemplates emailTemplateType = new EmailTemplateTypeWithoutTemplates();
+            // Set display name.
+            emailTemplateType.setDisplayName(templateType);
+            // Set id.
+            String templateTypeId = getEmailTemplateIdFromDisplayName(templateType);
+            emailTemplateType.setId(templateTypeId);
+            // Set location.
+            emailTemplateType.setSelf(getTemplateTypeLocation(templateTypeId));
+
+            if (requestedAttributeList != null) {
+                emailTemplateType.setTemplates(new ArrayList<>());
+            }
+            templateTypeMap.put(templateType, emailTemplateType);
+        }
 
-                templateTypeMap.put(emailTemplate.getTemplateType(), emailTemplateType);
+        // Populate optional email template information if exists.
+        if (requestedAttributeList != null) {
+            for (EmailTemplate emailTemplate : internalEmailTemplates) {
+                EmailTemplateWithID templateWithID = new EmailTemplateWithID();
+                for (String requestedAttribute : requestedAttributeList) {
+                    switch (requestedAttribute) {
+                        case Constants.TEMPLATES:
+                            templateWithID.setContentType(emailTemplate.getEmailContentType());
+                            templateWithID.setSubject(emailTemplate.getSubject());
+                            templateWithID.setBody(emailTemplate.getBody());
+                            templateWithID.setFooter(emailTemplate.getFooter());
+                            templateWithID.setId(emailTemplate.getLocale());
+                            break;
+                        case Constants.LOCALE:
+                            templateWithID.setId(emailTemplate.getLocale());
+                            break;
+                        case Constants.CONTENT_TYPE:
+                            templateWithID.setContentType(emailTemplate.getEmailContentType());
+                            break;
+                        case Constants.SUBJECT:
+                            templateWithID.setSubject(emailTemplate.getSubject());
+                            break;
+                        case Constants.BODY:
+                            templateWithID.setBody(emailTemplate.getBody());
+                            break;
+                        case Constants.FOOTER:
+                            templateWithID.setFooter(emailTemplate.getFooter());
+                            break;
+                    }
+                }
+                // Email template's display name is used to search templateTypeMap key because
```

Comment:
let's use multi-line comment format

Reply:
addressed in 1bbab048df7ab95cbf4757e21035235a9238d69e